engines/sci/graphics/cursor.cpp

```diff
@@ -538,7 +538,25 @@ void GfxCursor::kernelSetMacCursor(GuiResourceId viewNum, int loopNum, int celNu
 		return;
 	}
 
-	CursorMan.replaceCursor(macCursor);
+	if (_upscaledHires == GFX_SCREEN_UPSCALED_640x400) {
+		// Scale cursor by 2x
+		uint16 width = macCursor->getWidth() * 2;
+		uint16 height = macCursor->getHeight() * 2;
+		uint16 cursorHotspotX = macCursor->getHotspotX() * 2;
+		uint16 cursorHotspotY = macCursor->getHotspotY() * 2;
+		Common::SpanOwner<SciSpan<byte> > cursorBitmap;
+		cursorBitmap->allocate(width * height, "upscaled cursor bitmap");
+		SciSpan<const byte> sourceBitmap(macCursor->getSurface(), macCursor->getWidth() * macCursor->getHeight());
+		_screen->scale2x(sourceBitmap, *cursorBitmap, macCursor->getWidth(), macCursor->getHeight());
```

Comment:
Since we will now have 3 places in cursor.cpp where we scale the cursor, would it be possible to reuse this cursor scaling code?

Reply:
As I recall (I wrote this in March), I tried and failed to do that. Each section was always using different classes for its abstractions or operating under different constraints that were incompatible with existing utility code. You can probably find even more instances of manual scaling in SCI because of this. I couldn't make the pieces fit. Show me the light! =)

Reply:
OK. We could look for a possible way to do this after this PR is merged.